source/app/tests/test_configuration_module.py

```diff
@@ -0,0 +1,311 @@
+                "description": "Retain running",
+                "periods": {"office-hours"},
+                "name": "running",
+                "type": "schedule",
+            },
+            {
+                "timezone": "US/Pacific",
+                "description": "Office hours in Seattle (Pacific)",
+                "periods": {"office-hours"},
+                "name": "seattle-office-hours",
+                "type": "schedule",
+            },
+            {"description": "Instances stopped", "name": "stopped", "type": "schedule"},
+            {"periods": {"test3"}, "name": "test-error", "type": "schedule"},
+            {
+                "timezone": "US/Eastern",
+                "periods": {"sat3", "sat4"},
+                "name": "test-sat4",
+                "type": "schedule",
+            },
+            {
+                "timezone": "Asia/Macau",
+                "periods": {"test12", "test1"},
+                "name": "test-ssm1",
+                "type": "schedule",
+            },
+            {"periods": {"test1"}, "name": "test-ssm2", "type": "schedule"},
+            {
+                "description": "Testing config sun#4",
+                "periods": {"sun4", "sun4-2"},
+                "name": "test-sun4",
+                "type": "schedule",
+            },
+            {
+                "timezone": "US/Eastern",
+                "periods": {"test1"},
+                "name": "testMetrics",
+                "type": "schedule",
+            },
+            {
+                "timezone": "Europe/London",
+                "description": "Office hours in UK",
+                "periods": {"office-hours"},
+                "name": "uk-office-hours",
+                "type": "schedule",
+            },
+        ],
+    }
+    response = SchedulerConfigBuilder(logger=logger).build(config_data)
+    assert response.scheduled_services == {"ec2"}
+    assert response.default_timezone == "US/Eastern"
+    assert response.tag_name == "Schedule"
+    assert response.create_rds_snapshot == False
+    assert response.regions == {"us-east-1"}
+    assert response.remote_account_ids == []
+
+    for schedule_name in response.schedules:
+        if schedule_name == "docDbSchedule":
+            assert response.get_schedule(schedule_name).timezone == "US/Eastern"
+            for period in response.get_schedule(schedule_name).periods:
+                valid_current_datetime = datetime.datetime(
+                    2021, 6, 18, 14, 5, 59, 342380
+                )
+                assert (
+                    period.get("period").get_desired_state(
+                        logger, valid_current_datetime
+                    )
+                    == "running"
+                )
+
+                invalid_datetime = datetime.datetime(2021, 6, 20, 17, 30, 59, 34258)
+                assert (
+                    period.get("period").get_desired_state(logger, invalid_datetime)
+                    == "stopped"
+                )
+
```

Comment:
Having them together can provide value as well, to validate that it can make multiple scheduling decisions. but I agree that when the test fails it should be for one reason.